content/blog/cloud-native-secret-management-with-pulumi-esc-and-external-secrets-operator/index.md

```diff
@@ -0,0 +1,450 @@
+
+- Automate secret rotation by providing a way to rotate secrets automatically by using the external secret management
+  that supports secret rotation.
+
+## External Secrets Operator Architecture
+
+Central part of the [ESO architecture](https://external-secrets.io/latest/introduction/overview/) is the extension of
+the Kubernetes API with a
+new [Custom Resource Definition (CRD)](https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/)
+which defines where the secrets are stored and how they are synchronized into Kubernetes secrets. The ESO controller
+then retrieves the secrets from the external secret management system and creates or updates the Kubernetes secrets.
+Every time a secret is updated in the external secret management system, the ESO controller will synchronize the secret
+in the Kubernetes cluster and update the secret accordingly.
+
+![ESO Architecture](arch1.png)
+
+## Core Resources of ESO in a glance:
+
+- `(Cluster)SecretStore`: Defines the external secret management system where the secrets are stored. It contains the
+  necessary information to connect to the external secret management system like the endpoint, authentication
+  credentials, and other configuration options. The `SecretStore` is namespaced while the `ClusterSecretStore` is
+  cluster wide, which allows you cross-namespaced access to the secrets.
+
+- `(Cluster)ExternalSecret`: Defines the secret that needs to be synchronized into the Kubernetes cluster. It contains
+  the reference to the `SecretStore` and the secret key that needs to be synchronized and how the data should be
+  transformed and saved as a Kubernetes secret.
+
+- `PushSecret`: With the PushSecret resource, you can push secrets from the Kubernetes cluster to the external secret
+  management system. The Pulumi ESC provider for ESO supports this feature, which allows you to push secrets from the
+  cluster into Pulumi ESC. This feature is useful when you created secrets through other means and want to synchronize
+  them into Pulumi ESC to manage them centrally.
+
+## How to Use Pulumi ESC and ESO Together
+
+To demonstrate how to use Pulumi ESC and ESO together, we will create a simple example where we will store different
+secretes and configurations in Pulumi ESC and synchronize them into a Kubernetes cluster using ESO.
+
+To follow along, you need to have the following prerequisites:
+
+- A Kubernetes cluster (I will be using a local kind cluster, but you can use any Kubernetes cluster)
+- Pulumi CLI installed
+- `kubectl` CLI installed for some debugging
+
+### Step 1: Install ESO and Link to Pulumi ESC
+
+We will use Pulumi with the `pulumi-kuernetes` provider to deploy ESO into the Kubernetes cluster. As we need to provide
+a PAT (Personal Access Token) to authenticate with Pulumi ESC, we will create a secret in the Kubernetes cluster to
+store. But this secret will be retrieved from Pulumi ESC using the Pulumi ESC integration of Pulumi IaC.
```

Comment:
Might need to explain this more clearly... something like:

"We will be using ESC in two ways: First, to supply the Pulumi access token to ESO, via Pulumi IaC, which will allow the new ESO instance to access ESC later. Second, to provide configuration and secrets from ESC, via ESO, to the PodInfo application within the Kubernetes cluster."